examples/hosts/app-integration/schema-upgrade/src/app.tsx

```diff
@@ -0,0 +1,134 @@
+document.title = documentId;
+
+async function getInventoryListFromContainer(container: IContainer): Promise<IInventoryList> {
+    // Since we're using a ContainerRuntimeFactoryWithDefaultDataStore, our inventory list is available at the URL "/".
+    return requestFluidObject<IInventoryList>(container, { url: "/" });
+}
+
+async function getContainerKillBitFromContainer(container: IContainer): Promise<IContainerKillBit> {
+    // Our kill bit is available at the URL containerKillBitId.
+    return requestFluidObject<IContainerKillBit>(container, { url: containerKillBitId });
+}
+
+async function start(): Promise<void> {
+    const tinyliciousService = new TinyliciousService();
+
+    const module = { fluidExport: new InventoryListContainerRuntimeFactory() };
+    const codeLoader = { load: async () => module };
+
+    const loader = new Loader({
+        urlResolver: tinyliciousService.urlResolver,
+        documentServiceFactory: tinyliciousService.documentServiceFactory,
+        codeLoader,
+    });
+
+    let fetchedData: string | undefined;
+    let container: IContainer;
+    let inventoryList: IInventoryList;
+    let containerKillBit: IContainerKillBit;
+
+    if (createNew) {
+        fetchedData = await fetchData();
+        container = await loader.createDetachedContainer({ package: "no-dynamic-package", config: {} });
+        inventoryList = await getInventoryListFromContainer(container);
+        containerKillBit = await getContainerKillBitFromContainer(container);
+        await applyStringData(inventoryList, fetchedData);
+        await container.attach({ url: documentId });
+    } else {
+        container = await loader.resolve({ url: documentId });
+        containerKillBit = await getContainerKillBitFromContainer(container);
+        inventoryList = await getInventoryListFromContainer(container);
+    }
+
+    const writeToExternalStorage = async () => {
+        // CONSIDER: it's perhaps more-correct to spawn a new client to extract with (to avoid local changes).
+        // This can be done by making a loader.request() call with appropriate headers (same as we do for the
+        // summarizing client).  E.g.
+        // const exportContainer = await loader.resolve(...);
+        // const inventoryList = (await exportContainer.request(...)).value;
+        // const stringData = extractStringData(inventoryList);
+        // exportContainer.close();
+
+        const stringData = await extractStringData(inventoryList);
+        await writeData(stringData);
+
+        // Normally would be a void, we return the string here for demo purposes only.
+        return stringData;
+    };
+
+    const saveAndEndSession = async () => {
+        if (!containerKillBit.markedForDestruction) {
+            await containerKillBit.markForDestruction();
+        }
+
+        if (containerKillBit.dead) {
+            return;
+        }
+
+        // After the quorum proposal is accepted, our system doesn't allow further edits to the string
```

Comment:
This comment does not match implementation. markForDestruction() uses crc and does not wait for consensus, it only checks that write by this client happened. But we might still see ops coming in from other clients with reference seq # below this crc op, as those clients did not (yet) see this crc op when generating their ops.

I do not know if it changes anything (in terms of logical flow of this code), but comment is wrong and should be removed IMHO.